### ISSUE TITLE
Lit + Jest + Typescript + SWC Jest

### ISSUE THREAD

**Issue by gethari** [lit/lit#4580]
### Which package(s) are affected?

Lit Core (lit / lit-html / lit-element / reactive-element)

### Description

I am trying to use [@swc/jest](https://swc.rs/docs/usage/jest) to speed up my tests on my `lit+typescript+jest` setup. When using the `swc/jest` as the test runner the `willUpdate` method is not getting executed. From what I see/find the call stack is different when using `ts-jest` & it works fine, but on using `swc/jest` the callstack is missing the `_callSuper()`.

I am attaching a simple repository that can reproduce the issue. Additionally, I've included screenshots comparing the behavior with swc and ts-jest for your reference.

### swc
![image](https://github.com/lit/lit/assets/20477956/b171e73f-8c21-4b6f-80df-54cc17dd2e41)

### ts-jest
![image](https://github.com/lit/lit/assets/20477956/bda8b731-38f3-4c3c-9a26-1e2780da0855)

### Reproduction

Please check the readme in this repository for detailed repro steps.
https://github.com/gethari/typescript-webpack-lit-html-jest-swc

### Workaround

I have not found a workaround

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

3.1.2

### Browser/OS/Node environment

Browser: Chrome 103.0.5060.53
OS: Windows 11
Node version: 20.11.1
npm version: 10.2.4

**Comment by augustjk** [comment]
Are you using `@property` decorators? It might be that SWC isn't compiling them right. I don't know if SWC supports reading project tsconfigs but you should make sure SWC configs match those of tsconfig listed here https://lit.dev/docs/components/decorators/#decorators-typescript

**Comment by gethari** [comment]
@augustjk thanks for responding, This is my [tsconfig.json](https://github.com/gethari/typescript-webpack-lit-html-jest-swc/blob/f7c6e742c346b7171f5b8d0a58fd8028c5cdeada/tsconfig.json#L4-L5) , it has both the properties mentioned in the `lit` docs, as well my `.swcrc` config has [decorators](https://github.com/gethari/typescript-webpack-lit-html-jest-swc/blob/f7c6e742c346b7171f5b8d0a58fd8028c5cdeada/.swcrc#L7) enabled.

I am not sure about the actual cause of the issue as I'm trying to use 3 different things together, that nobody before have seem to done ( so far from my search).  Hence why raised issue for help across all 3 repo.

**Comment by augustjk** [comment]
My suspicion is that swc's decorator config is the standard decorator proposal implementation, not TypeScript's experimental decorator, which might be this https://swc.rs/docs/configuration/compilation#jsctransformlegacydecorator

**Comment by gethari** [comment]
@augustjk you are right, by making the below config changes to `swcrc` I was able to bypass the error & got the tests up, but it results in a new error on a test, that is testing a component that is built on [Stencil](https://stenciljs.com/)

```
Cannot find module 'web-date-picker/dist/types/components/duet-date-picker/duet-date-picker' from 'src/components/date-picker/date-picker.component.ts
```

![image](https://github.com/lit/lit/assets/20477956/baa700a1-bebd-44d5-8371-6e05a78ca5bf)


![image](https://github.com/lit/lit/assets/20477956/89a2d9a6-7499-489a-84be-159f5a54c39e)


But when I make the below props to false, the `date-picker` tests are working

```json
"legacyDecorator": true,
"decoratorMetadata": true,
```

**Comment by gethari** [comment]
I also found these, I think might be related

- https://github.com/swc-project/swc/issues/6833
- https://github.com/swc-project/swc/issues/6824

my `swcrc` config for ref

```
{
    "$schema": "https://json.schemastore.org/swcrc",
    "jsc": {
        "parser": {
            "syntax": "typescript",
            "dynamicImport": true,
            "decorators": true
        },
        "baseUrl": ".",
        "paths": {
            "*": ["src/*"]
        },
        "keepClassNames": true,
        "transform": {
            "legacyDecorator": true,
            "decoratorMetadata": true,
            "useDefineForClassFields": false
        },
        "target": "esnext"
    }
}
```

**Comment by augustjk** [comment]
I don't know about Stencil but we don't require `decoratorMetadata` to be true.
